lib/src/cmaf/ham/services/validate.ts

```diff
@@ -0,0 +1,455 @@
+import {
+	AudioTrack,
+	Presentation,
+	SelectionSet,
+	SwitchingSet,
+	Track,
+	VideoTrack,
+	TextTrack,
+	Segment,
+} from '../types/model';
+
+// TODO: Track and Segment validation are so simple for the PoC. The method should validate more deeper and connecting the things
+// in the model like durations, urls, frame rate, etc.
+
+/**
+ * CMAF-HAM Validation type
+ *
+ * @group CMAF
+ *
+ * @alpha
+ */
+type Validation = {
+	status: boolean;
+	errorMessages: string[];
+};
+
+/**
+ * Validate a presentation.
+ * It validates in cascade, calling each child validation method.
+ *
+ * @param presentation - Presentation from cmaf ham model
+ *
+ * @returns Validation
+ *
+ * @group CMAF
+ *
+ * @alpha
+ *
+ */
+function validatePresentation(presentation: Presentation): Validation {
+	const validation: Validation = { status: true, errorMessages: [] };
+
+	if (!presentation.id) {
+		validation.status = false;
+		validation.errorMessages.push('Presentation id is undefined');
+	}
+
+	validateSelectionSets(
+		presentation.selectionSets,
+		presentation.id,
+		validation,
+	);
+
+	return validation;
+}
+
+/**
+ * Validate a list of selection set.
+ * It validates in cascade, calling each child validation method.
+ *
+ * @param selectionSets - List of SelectionSet from cmaf ham model
+ * @param presentationId - Optional: parent presentation id
+ *
+ * @returns Validation
+ *
+ * @group CMAF
+ *
+ * @alpha
+ *
+ */
+function validateSelectionSets(
+	selectionSets: SelectionSet[],
+	presentationId?: string,
+	prevValidation?: Validation,
```

Comment:
`prevValidation` is not defined on jsocs, this applies to all code occurencies

Reply:
Done!